NitroxClient/GameLogic/Fires.cs

```diff
@@ -0,0 +1,149 @@
+            this.packetSender = packetSender;
+        }
+
+        /// <summary>
+        /// Triggered when <see cref="SubFire.CreateFire(SubFire.RoomFire)"/> is executed. To create a new fire manually, 
+        /// call <see cref="Create(string, Optional{string}, Optional{CyclopsRooms}, Optional{int})"/>
+        /// </summary>
+        public void OnCreate(Fire fire, SubFire.RoomFire room, int nodeIndex)
+        {
+            Optional<string> subRootGuid = Optional<string>.Of(GuidHelper.GetGuid(fire.fireSubRoot.gameObject));
+            // Optional<CyclopsRooms> startInRoom = Optional<CyclopsRooms>.Of(room.roomLinks.room);
+            // Optional<int> activeNodeIndex = Optional<int>.Of(nodeIndex);
+
+            FireCreated packet = new FireCreated(GuidHelper.GetGuid(fire.gameObject), subRootGuid, room.roomLinks.room, nodeIndex);
+            packetSender.Send(packet);
+        }
+
+        /// <summary>
+        /// Triggered when <see cref="Fire.Douse(float)"/> is executed. To Douse a fire manually, retrieve the <see cref="Fire"/> call the Douse method
+        /// </summary>
+        public void OnDouse(Fire fire, float douseAmount)
+        {
+            string fireGuid = GuidHelper.GetGuid(fire.gameObject);
+
+            // Temporary packet limiter
+            if (!fireDouseAmount.ContainsKey(fireGuid))
+            {
+                fireDouseAmount.Add(fireGuid, douseAmount);
+            }
+            else
+            {
+                float summedDouseAmount = fireDouseAmount[fireGuid] + douseAmount;
+
+                if (summedDouseAmount > FIRE_DOUSE_AMOUNT_TRIGGER)
+                {
+                    // It is significantly faster to keep the key as a 0 value than to remove it and re-add it later.
+                    fireDouseAmount[fireGuid] = 0;
+
+                    FireDoused packet = new FireDoused(fireGuid, douseAmount);
+                    packetSender.Send(packet);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Create a new <see cref="Fire"/>. Majority of code copied from <see cref="SubFire.CreateFire(SubFire.RoomFire)"/>. Currently does not support Fires created outside of a Cyclops
+        /// </summary>
+        /// <param name="fireGuid">Guid of the Fire. Used for identification when dousing the fire</param>
+        /// <param name="subRootGuid">Guid of the Cyclops <see cref="SubRoot"/></param>
+        /// <param name="room">The room the Fire will be spawned in</param>
+        /// <param name="spawnNodeIndex">Each <see cref="CyclopsRooms"/> has multiple static Fire spawn points called spawnNodes. If the wrong index is provided,
+        ///     the clients will see fires in different places from the owner</param>
+        public void Create(string fireGuid, Optional<string> subRootGuid, CyclopsRooms room, int spawnNodeIndex)
+        {
+            // A Fire is either on a Cyclops or not. Currently, creating a Fire is only possible if it's in a Cyclops.
```

Comment:
Is this function even reusable for other fires?  It takes in a CyclopsRoom

Reply:
I removed all of the non-Cyclops checks. All of the fire creation code is specific to the Cyclops.